tests/unit/conditions/test_auth.py

```diff
@@ -0,0 +1,112 @@
+import pytest
+
+from nucypher.policy.conditions.auth import Auth, EIP712Auth, SIWEAuth
```

Comment:
Help me to build a mental picture or system metaphor for this new API:  what is the relationship between the `Auth` ABC tree and say, Ciphertext headers?

Reply:
Nothing - this PR is Bob-specific at the moment. the `auth` is related to the authentication/verification of what is provided in `:userAddress`. One potential follow-on thought is that we could add additional auth checks like "confirming the domain allowed via SIWE" eg. what lit does - https://developer.litprotocol.com/v2/accesscontrol/evm/siwe#domain.

This type of auth condition could be made more generally as something like, "check an additional auth property" as an additional condition on the ciphertext:

```json
{
    "conditionType": "auth",
    "scheme": "SIWE",
    "authMessagePropertyToVerify": "domain"  // not a real name just being overly wordy for clarity
    "returnValueTest": {
      comparator: "==",
      value: "localhost:3050",
    },
}
```
but this is still up in the air and needs adopter validation.

Reply:
Agreed, let's leave the allowed domain confirmation until after we have some adopters of the joint SDK, since we don't know exactly what other auth properties might be requested, and we may still want to combine these more closely with conditions.